src/include/network/connection_handle_factory.h

```diff
@@ -0,0 +1,55 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Terrier
+//
+// connection_handle_factory.h
+//
+// Identification: src/include/network/connection_handle_factory.h
+//
+// Copyright (c) 2015-2018, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <unordered_map>
+
+#include "network/connection_handle.h"
+#include "network/terrier_server.h"
+
+namespace terrier {
+namespace network {
+
+/**
+ * @brief Factory class for constructing ConnectionHandle objects
+ * Each ConnectionHandle is associated with read and write buffers that are
+ * expensive to reallocate on the fly. Thus, instead of destroying these wrapper
+ * objects when they are out of scope, we save them until we can transfer their
+ * buffers to other wrappers.
+ */
+// TODO(Tianyu): Additionally, it is hard to make sure the ConnectionHandles
+// don't leak without this factory since they are essentially managed by
+// libevent if nothing in our system holds reference to them, and libevent
+// doesn't cleanup raw pointers.
+class ConnectionHandleFactory {
```

Comment:
We will eventually want to replace this with the general purpose object pool from terrier. You don't have to do it in this PR though.